src/accessibility/RovingTabIndex.tsx

```diff
@@ -270,9 +270,9 @@ export const RovingTabIndexProvider: React.FC<IProps> = ({
 // onFocus should be called when the index gained focus in any manner
 // isActive should be used to set tabIndex in a manner such as `tabIndex={isActive ? 0 : -1}`
 // ref should be passed to a DOM node which will be used for DOM compareDocumentPosition
-export const useRovingTabIndex = (inputRef?: Ref): [FocusHandler, boolean, Ref] => {
+export const useRovingTabIndex = <T extends HTMLElement, >(inputRef?: RefObject<T>): [FocusHandler, boolean, RefObject<T>] => {
```

Comment:
We could copy [`useContextMenu`](https://github.com/matrix-org/matrix-react-sdk/blob/132e19beb77d89d70c08c9d831f950bb0019d7d6/src/components/structures/ContextMenu.tsx#L521)

```
<T extends any = HTMLElement>(): ContextMenuTuple<T> => {
```

Reply:
hmm, it results in type errors and I'm not confident in which is the better solution. Feedback welcome here!

Reply:
How best to split this line up? It's too long

Reply:
Yup!